### ISSUE TITLE
API for pulse counting

### ISSUE THREAD

**Issue by ytai** [ytai/ioio#20]
Either in conjunction with PulseInput or with DigitalInput.

The latter can be implemented on the Android-side, which is less efficient but easier. Probably not good enough if we want to support high pulse rates.

Maybe combine this into a full-blown encoder API.

**Comment by jfwork** [comment]
Hi Ytai!

First of all, amazing work! ;)

Second of all, when do you think this feature could be implemented? I'm currently working on a project that controls servo motors and needs to count steps from Hall sensor feedback. Frequency is hi (I think it is around 1/16ms) so imo it would be easier to implement it at low level (IOIO).

I've found a place where you start counting rising edges and where you prepare data for Java client. I could try to play with it and flash IOIO board with my changes but I would need your help not to screw things up. If you could find some time to discuss it? :)

Denis

**Comment by ytai** [comment]
Yeah, I can help you if you want. A good start would be to make sure you
can build, install and run the existing firmware. Then, you can write up a
quick proposal of how you're thinking to implement this, which I can review.
